bundles/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/GroupRenderer.java

```diff
@@ -36,16 +38,27 @@ public boolean canRender(Widget w) {
      */
     @Override
     public EList<Widget> renderWidget(Widget w, StringBuilder sb) throws RenderException {
-        String snippet = getSnippet("group");
 
-        snippet = StringUtils.replace(snippet, "%id%", itemUIRegistry.getWidgetId(w));
-        snippet = StringUtils.replace(snippet, "%icon%", escapeURLPath(itemUIRegistry.getIcon(w)));
-        snippet = StringUtils.replace(snippet, "%label%", getLabel(w));
+        if (((Group) w).isExpanded()) {
+            EList<Widget> children = itemUIRegistry.getChildren((LinkableWidget) w);
 
-        // Process the color tags
-        snippet = processColor(w, snippet);
+            TextRenderer textRenderer = new TextRenderer();
+            textRenderer.setItemUIRegistry(itemUIRegistry);
+            
+            for (Widget child : children) 
+                textRenderer.renderWidget(child, sb);            
```

Comment:
Why is the text renderer used for every child? Adding a "expanded" to the DemoSwitchGroup in the demo.sitemap, results immediately in an exception here:
```
16:32:11.925 WARN  o.e.j.servlet.ServletHandler[:561] - /classicui/app
java.lang.ClassCastException: org.eclipse.smarthome.model.sitemap.impl.SwitchImpl cannot be cast to org.eclipse.smarthome.model.sitemap.Text
	at org.eclipse.smarthome.ui.classic.internal.render.TextRenderer.renderWidget(TextRenderer.java:39)
	at org.eclipse.smarthome.ui.classic.internal.render.GroupRenderer.renderWidget(GroupRenderer.java:49)
	at org.eclipse.smarthome.ui.classic.internal.render.PageRenderer.renderWidget(PageRenderer.java:162)
	at org.eclipse.smarthome.ui.classic.internal.render.PageRenderer.processChildren(PageRenderer.java:120)
	at org.eclipse.smarthome.ui.classic.internal.render.PageRenderer.processChildren(PageRenderer.java:141)
	at org.eclipse.smarthome.ui.classic.internal.render.PageRenderer.processPage(PageRenderer.java:84)
	at org.eclipse.smarthome.ui.classic.internal.servlet.WebAppServlet.service(WebAppServlet.java:145)
	at org.eclipse.equinox.http.servlet.internal.ServletRegistration.service(ServletRegistration.java:61)
	at org.eclipse.equinox.http.servlet.internal.ProxyServlet.processAlias(ProxyServlet.java:128)
	at org.eclipse.equinox.http.servlet.internal.ProxyServlet.service(ProxyServlet.java:60)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
```